BlueprintUI/Sources/Layout/EqualSizeStack.swift

```diff
@@ -0,0 +1,134 @@
+    }
+
+}
+
+extension EqualSizeStack {
+
+    public enum Direction {
+        case vertical
+        case horizontal
+    }
+
+}
+
+extension EqualSizeStack {
+
+    /// A layout that sizes all of its children equally, stacking them according to
+    /// the specified `direction` and spacing them according to the specified `gutter`.
+    fileprivate struct EqualSizeLayout: Layout {
+
+        /// The direction in which this layout will stack its children.
+        var direction: Direction
+
+        /// The amount of space between children in this layout.
+        var gutter: CGFloat
+
+        /// - Parameters:
+        ///     - direction: The direction in which this layout will stack its children.
+        ///     - gutter: The amount of space between children in this layout.
+        init(direction: Direction, gutter: CGFloat) {
+            self.direction = direction
+            self.gutter = gutter
+        }
+
+        func measure(in constraint: SizeConstraint, items: [(traits: Void, content: Measurable)]) -> CGSize {
+            var totalSize = items
+                .map { $1.measure(in: constraint) }
+                .reduce(into: CGSize.zero) { totalSize, itemSize in
+                    switch direction {
+                    case .horizontal:
+                        totalSize.width += itemSize.width
+                        totalSize.height = max(totalSize.height, itemSize.height)
+                    case .vertical:
+                        totalSize.width = max(totalSize.width, itemSize.width)
+                        totalSize.height += itemSize.height
+                    }
+            }
+
+            switch direction {
+            case .horizontal:
+                totalSize.width += gutter * CGFloat(items.count - 1)
+            case .vertical:
+                totalSize.height += gutter * CGFloat(items.count - 1)
+            }
+
+            return totalSize
+        }
+
+        func layout(size: CGSize, items: [(traits: (), content: Measurable)]) -> [LayoutAttributes] {
+            guard items.count > 0 else { return [] }
+
+            let baseSize: CGSize
+            switch direction {
+            case .horizontal:
+                baseSize = CGSize(
+                    width: (size.width - (gutter * CGFloat(items.count - 1))) / CGFloat(items.count),
+                    height: size.height)
+            case .vertical:
+                baseSize = CGSize(
+                    width: size.width,
+                    height: (size.height - (gutter * CGFloat(items.count - 1))) / CGFloat(items.count))
+            }
+
+            return Array(repeating: LayoutAttributes(size: baseSize), count: items.count)
+                .enumerated()
+                .map { index, attributes in
+                    var attributes = attributes
+                    switch direction {
+                    case .horizontal:
+                        attributes.frame.origin.x = (baseSize.width + gutter) * CGFloat(index)
```

Comment:
How do you feel about rounding this value to the nearest pixel?

Reply:
+1

Reply:
Same for all the things in here, likely.